doc/releases/release-notes-3.6.rst

```diff
@@ -573,38 +606,183 @@ Networking
+    Update period regardless of the lifetime value.
+  * Fixed composite read/write access rights check.
+  * Added shell command to delete object and resource instances.
+  * Fixed a bug in block-wise transfer, where block-wise ACKs were sent with
+    wrong response code.
+  * Fixed object version reporting for LwM2M version 1.1.
+  * Added support for DTLS Connection Identifier in LwM2M engine.
+  * Added support for LwM2M Server Disable executable resource.
+  * Implemented fallback mechanism for LwM2M server selection during registration
+    phase. The engine will now try to choose different server if the current one
+    becomes unavailable or disabled.
+  * Added support for storing LwM2M error list in settings.
+  * Fixed pmin observer attribute handling in tickless mode.
+  * Added support for notifying the application about ongoing CoAP transmissions
+    with ``set_socket_state()`` callback.
+  * Deprecated unsigned 64-bit integer value type, as it's not represented in the spec.
+    Use signed 64-bit integer instead.
+  * Added a callback for LwM2M Gateway object, which allows to handle LwM2M messages
+    with prefixed path.
+  * Added LwM2M-specific macros for object initialization during boot.
+  * Several other minor bugfixes ans improvements.
 
-  * It is now possible to have separate IPv4 TTL value and IPv6 hop limit value for
-    unicast and multicast packets. This can be controlled in each socket via
-    :c:func:`setsockopt` API.
+* Misc:
 
   * Added support for compile time network event handlers using the macro
     :c:macro:`NET_MGMT_REGISTER_EVENT_HANDLER`.
-
   * The :kconfig:option:`CONFIG_NET_MGMT_EVENT_WORKER` choice is added to
     allow emitting network events using the system work queue or synchronously.
+  * Removed redundant Network Connectivity API documentation page.
+  * Improved thread safety fo the network connections subsystem.
+  * Removed ``eth_native_posix`` sample.
+  * Removed redundant ``arb`` and ``fv2015`` fields  from
+    :c:struct:`net_pkt_cb_ieee802154`.
+  * Introduced a separate mutex for TX at the network interface level, to prevent
+    concurrent access on TX to drivers that are not re-entrant.
+  * Fixed netmask not being registered for loopback address.
+  * Added support for binding to a specific network interface at the net_context
+    level.
+  * Added IGMPv3 support.
+  * Added a new network event, ``NET_EVENT_L4_CMD_HOSTNAME_CHANGED``, triggered
+    upon hostname change.
+  * Refactored net_context option getters/setters to reduce code duplication.
+  * Fixed a possible packet leak at the ARP level, in case of errors during ARP
+    packet creation.
+  * Added support for analyzing SNTP time uncertainty.
+  * Fixed network interface being brought up even when underlying device is not
+    ready.
+  * Added start/stop functions for dummy interfaces.
+  * Added a detailed network configuration guide to the documentation.
```

Comment:
I would link to it, it is very useful resource and making it easier to not miss it can only be a good thing.

Reply:
Good point, added a link to the guide, now need to wait for the doc build to see if I didn't mess up.